GeneratorInterface/GenFilters/plugins/AJJGenJetFilter.cc

```diff
@@ -0,0 +1,241 @@
+
+      maxPhotonEta(iConfig.getParameter<double>("maxPhotonEta")),
+      minPhotonPt(iConfig.getParameter<double>("minPhotonPt")),
+      maxPhotonPt(iConfig.getParameter<double>("maxPhotonPt")) {
+  m_GenParticleCollection = consumes<reco::GenParticleCollection>(iConfig.getParameter<edm::InputTag>("genParticles"));
+
+  if (ptMin > 0) {
+    m_GenJetCollection = consumes<reco::GenJetCollection>(iConfig.getParameter<edm::InputTag>("GenJetCollection"));
+  }
+  edm::LogInfo("AJJGenJetFilter") << "Parameters:"
+                                  << "jPtMin:" << ptMin << ",jEta:" << etaMin << "--" << etaMax << ",minDR(j,lep)"
+                                  << deltaRJetLep << ",deltaEta(j1,j2)" << minDeltaEta << "--" << maxDeltaEta
+                                  << "m(j1,j2) < " << mininvmass << "PhotonPt" << minPhotonPt << "--" << maxPhotonPt
+                                  << "PhotonEta" << maxPhotonEta;
+}
+
+AJJGenJetFilter::~AJJGenJetFilter() {}
+
+const vector<const reco::GenParticle*> AJJGenJetFilter::filterGenLeptons(
+    const vector<reco::GenParticle>* particles) const {
+  vector<const reco::GenParticle*> out;
+
+  for (const auto& p : *particles) {
+    int absPdgId = std::abs(p.pdgId());
+
+    if (((absPdgId == 11) || (absPdgId == 13) || (absPdgId == 15)) && p.isHardProcess()) {
+      out.push_back(&p);
+    }
+  }
+  return out;
+}
+
+const vector<const reco::GenParticle*> AJJGenJetFilter::filterGenPhotons(
+    const vector<reco::GenParticle>* particles) const {
+  vector<const reco::GenParticle*> out;
+
+  for (const auto& p : *particles) {
+    int absPdgId = std::abs(p.pdgId());
+
+    if ((absPdgId == 22) && p.isHardProcess()) {
+      if (std::abs(p.eta()) < maxPhotonEta && p.pt() > minPhotonPt && p.pt() <= maxPhotonPt) {
+        out.push_back(&p);
+      } else {
+        edm::LogInfo("AJJPhoton") << "photon rejected, pt:" << p.pt() << " , eta:" << p.eta();
+      }
+    }
+  }
+
+  return out;
+}
+
+const vector<const reco::GenJet*> AJJGenJetFilter::filterGenJets(const vector<reco::GenJet>* jets) const {
+  vector<const reco::GenJet*> out;
+
+  for (auto const& j : *jets) {
+    if (j.p4().pt() > ptMin && j.p4().eta() > etaMin && j.p4().eta() < etaMax) {
+      out.push_back(&j);
+    } else {
+      edm::LogInfo("AJJJets") << "Jet rejected, pt:" << j.p4().pt() << " eta:" << j.p4().eta();
+    }
+  }
+
+  return out;
+}
+
+// ------------ method called to skim the data  ------------
+bool AJJGenJetFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const {
+  using namespace edm;
+  // Getting filtered generator jets
+
+  //Handle<reco::GenParticleCollection> genParticelesCollection;
+  //iEvent.getByToken(m_GenParticleCollection, genParticelesCollection);
+  //const vector<reco::GenParticle>* genParticles = genParticelesCollection.product();
+  auto const& genParticles = iEvent.get(m_GenParticleCollection);
+  vector<const reco::GenParticle*> filGenLep = filterGenLeptons(&genParticles);
```

Comment:
Since you only check the size of the `filGenPhotons`, this line should be more conveniently moved after L182